Replacing Postgres deployment instructions with Clickhouse ones (#1701)
PostHog/posthog.com: contents/docs/self-host/deploy/aws.md

    - aws
---

> First we need to set up a Kubernetes Cluster, see [Setup EKS - eksctl](https://docs.aws.amazon.com/eks/latest/userguide/getting-started-eksctl.html). The default nodes (2x m5.large) work well for running Posthog.

Reviewer: ```suggestion
First, we need to set up a Kubernetes Cluster, see [Setup EKS - eksctl](https://docs.aws.amazon.com/eks/latest/userguide/getting-started-eksctl.html). The default nodes (2x m5.large) work well for running Posthog.
```

> Here's the minimal required `values.yaml` that we'll be using later. You can find an overview of the parameters that can be configured during installation under [configuration](/docs/self-host/deploy/configuration).

Reviewer: ```suggestion
Here's the minimal required `values.yaml` that we'll be using later. You can find an overview of the parameters that can be configured during installation under [chart configuration](/docs/self-host/deploy/configuration).
```

```yaml
cloud: "aws"
ingress:
  hostname: <your-hostname>
  nginx:
    enabled: true
certManager:
  enabled: true
```

### Installing the chart

To install the chart with the release name `posthog` in `posthog` namespace, run the following:

```console
helm repo add posthog https://posthog.github.io/charts-clickhouse/
helm repo update
helm install -f values.yaml --timeout 20m --create-namespace --namespace posthog posthog posthog/posthog
```
    
### Lookup external IP

```console
kubectl get svc --namespace posthog posthog-ingress-nginx-controller
```
### Setting up DNS

> Create a `CNAME` record from your desired hostname to the external IP.

Author: But doesn't AWS gives `CNAME`'s? At least that's what I saw, if you check the awstest one in route 53

> Create a `CNAME` record from your desired hostname to the external IP.

Author: Maybe I should just say "Create a record" everywhere?

> Create a `CNAME` record from your desired hostname to the external IP.

Reviewer 2: Well the main thing is this:
Create a `CNAME` record from your desired hostname to the external IP.

You can't have a `CNAME` => `IP`
- `CNAME` => any dns hostname ✅ 
- `A` => `IP` ie `1.1.1.1` ✅
- `CNAME` => `1.1.1.1` ❌ 
- `CNAME` => `acefaafc147b54d759c4915328f3066d-376747404.eu-west-3.elb.amazonaws.com` ✅ 

So saying "Create a `CNAME` record for your desired hostname to the external IP." would be a bad thing.
It should say (for AWS) "Create a `CNAME` record for your desired hostname to the hostname associated with the AWS LB"

`acefaafc147b54d759c4915328f3066d-376747404.eu-west-3.elb.amazonaws.com` is a hostname (not an IP) so a `CNAME` is perfect here.

> Create a `CNAME` record from your desired hostname to the external IP.

Author: 🤦‍♀️  oh of course, I didn't read the sentence until the end

> Create a `CNAME` record from your desired hostname to the external IP.

Author: ```suggestion
Create a `CNAME` record from your desired hostname to the hostname associated with the AWS LB.
```

### I cannot connect to my PostHog instance after creation
As a troubleshooting tool, you can allow HTTP access by setting these values in your `values.yaml`, but we recommend always accessing PostHog via https.
```yaml
ingress:
  nginx:
    redirectToTLS: false
  letsencrypt: false
web:
  secureCookies: false
```

After upgrading you can run the following to get the IP to access Posthog.
```console
export INGRESS_IP=$(kubectl get --namespace posthog ingress posthog -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
echo "Visit http://$INGRESS_IP to use PostHog\!"
```

## Upgrading the chart

To upgrade the chart using [Helm >= v3](https://helm.sh/) with the release name `posthog` in `posthog` namespace, run the following:

```console
helm repo update
helm upgrade -f values.yaml --timeout 20m --namespace posthog posthog posthog/posthog
```

> See [the Helm docs](https://helm.sh/docs/helm/helm_upgrade/) for documentation on the `helm upgrade` command.

> When upgrading major versions, see [Upgrade notes](/docs/self-host/deploy/upgrade-notes)

Reviewer: ```suggestion
When upgrading major versions, see [Upgrade notes](/docs/self-host/deploy/upgrade-notes).
```

## Uninstalling the Chart

To uninstall the chart using [Helm >= v3](https://helm.sh/) with the release name `posthog` in `posthog` namespace, run the following:
```console
$ helm uninstall posthog --namespace posthog
```

> See [the Helm docs](https://helm.sh/docs/helm/helm_uninstall/) for documentation on the `helm uninstall` command.

> The command above removes all the Kubernetes components associated with the chart and deletes the release. Sometimes everything doesn't get properly removed, if that happens try deleting the namespace

Reviewer: ```suggestion
The command above removes all the Kubernetes components associated with the chart and deletes the release. Sometimes everything doesn't get properly removed. If that happens try deleting the namespace:
```
